service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/EhrExtractMapper.java

```diff
@@ -0,0 +1,85 @@
+package uk.nhs.adaptors.gp2gp.ehr;
+
+import java.util.Optional;
+
+import uk.nhs.adaptors.gp2gp.common.service.FhirParseService;
+import uk.nhs.adaptors.gp2gp.common.task.TaskIdService;
+import uk.nhs.adaptors.gp2gp.ehr.model.EhrExtractStatus;
+import uk.nhs.adaptors.gp2gp.ehr.model.EhrExtractTemplateParameters;
+import uk.nhs.adaptors.gp2gp.ehr.utils.TemplateUtils;
+import uk.nhs.adaptors.gp2gp.gpc.GetGpcStructuredTaskDefinition;
+
+import org.hl7.fhir.dstu3.model.Bundle;
+import org.hl7.fhir.dstu3.model.Patient;
+import org.hl7.fhir.dstu3.model.Reference;
+import org.hl7.fhir.dstu3.model.ResourceType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.github.mustachejava.Mustache;
+
+@Component
+public class EhrExtractMapper {
+    private static final Mustache EHR_EXTRACT_TEMPLATE = TemplateUtils.loadTemplate("ehr_extract_template.mustache");
+    private static final String SLASH = "/";
+
+    private final FhirParseService fhirParseService;
+    private final TaskIdService taskIdService;
+
+    @Autowired
+    public EhrExtractMapper(FhirParseService fhirParseService, TaskIdService taskIdService) {
+        this.fhirParseService = fhirParseService;
+        this.taskIdService = taskIdService;
+    }
+
+    public EhrExtractTemplateParameters mapJsonToEhrFhirExtractParams(EhrExtractStatus ehrExtractStatus,
```

Comment:
No `EhrExtractStatus ehrExtractStatus` here, everything needed should be contained in `GetGpcStructuredTaskDefinition`. If it is not there we need to add it in.